arborist/server.go

```diff
@@ -1129,6 +1167,42 @@ func (server *Server) handleUserGrantPolicy(w http.ResponseWriter, r *http.Reque
 	_ = jsonResponseFrom(nil, http.StatusNoContent).write(w, r)
 }
 
+func (server *Server) handleBulkUserGrantPolicy(w http.ResponseWriter, r *http.Request, body []byte) {
+	username := mux.Vars(r)["username"]
+	var requestPolicies []RequestPolicy
+	err := json.Unmarshal(body, &requestPolicies)
+	if err != nil {
+		msg := fmt.Sprintf("could not parse policy name in JSON: %s", err.Error())
+		server.logger.Info("tried to grant policy to user but input was invalid: %s", msg)
+		response := newErrorResponse(msg, 400, nil)
+		_ = response.write(w, r)
+		return
+	}
+
+	for _, requestPolicy := range requestPolicies {
+		var expiresAt *time.Time
```

Comment:
same comment about duplicated code in `handleUserGrantPolicy` and `handleBulkUserGrantPolicy`